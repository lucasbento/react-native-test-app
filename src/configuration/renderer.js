import React from 'react';
import {render} from 'react-native-testing-library';
import {Provider} from 'react-redux';
import store from '../store';

const customRender = (children, options = {}) =>
  render(<Provider store={store}>{children}</Provider>, options);

export * from 'react-native-testing-library';

export {customRender as render};
