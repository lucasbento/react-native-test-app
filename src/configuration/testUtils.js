const responses = {
  successful: ({data = undefined, ...rest} = {}) => ({
    status: 200,
    data,
    success: true,
    ...rest,
  }),
  failed: ({data = undefined, ...rest} = {}) => ({
    status: 500,
    data,
    success: false,
    ...rest,
  }),
};

export {responses};
