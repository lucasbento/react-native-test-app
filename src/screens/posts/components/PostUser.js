import React from 'react';
import PropTypes from 'prop-types';
import {Text, StyleSheet} from 'react-native';

const PostUser = ({name, username, company}) => (
  <Text style={styles.user}>
    By {name} (<Text style={styles.username}>@{username}</Text>) from{' '}
    {company.name}
  </Text>
);

PostUser.propTypes = {
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  company: PropTypes.shape({
    name: PropTypes.string,
  }).isRequired,
};

const styles = StyleSheet.create({
  user: {
    marginTop: 7,
    color: '#7E7E7E',
  },
  username: {
    color: '#00B2FF',
  },
});

export default PostUser;
