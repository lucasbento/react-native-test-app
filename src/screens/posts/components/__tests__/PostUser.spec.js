import React from 'react';
import {shallow} from '../../../../configuration/renderer';
import PostUser from '../PostUser';
import user from '../../../../../__fixtures__/user';

describe('<PostUser />', () => {
  const props = {
    name: user.name,
    username: user.username,
    company: user.company,
  };

  it('renders correctly', async () => {
    const {output} = shallow(<PostUser {...props} />);

    expect(output).toMatchSnapshot();
  });
});
