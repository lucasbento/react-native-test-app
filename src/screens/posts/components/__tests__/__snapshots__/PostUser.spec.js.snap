// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostUser /> renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#7E7E7E",
      "marginTop": 7,
    }
  }
>
  By 
  Leanne Graham
   (
  <Text
    style={
      Object {
        "color": "#00B2FF",
      }
    }
  >
    @
    Bret
  </Text>
  ) from
   
  Romaguera-Crona
</Text>
`;
